src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/maven/ReadMavenPomStep.java

```diff
@@ -123,7 +124,12 @@ protected Model run() throws Exception {
             if (path.isDirectory()) {
                 throw new FileNotFoundException(path.getRemote() + " is a directory.");
             }
-            return new MavenXpp3Reader().read(path.read());
+            
+            InputStream	stream = path.read();
```

Comment:
A try with resource is better to use, as done in #25 .
Here if an exception is thrown in `new MavenXpp3Reader().read(` the stream won't be closed and you're stuck witht he same problem.